server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/NewCoverageMeasuresStep.java

```diff
@@ -176,111 +164,12 @@ public NewLineCoverageFormula() {
-      super(
-        new SingleWithUncoveredMetricKeys(CoreMetrics.NEW_IT_LINES_TO_COVER_KEY, CoreMetrics.NEW_IT_UNCOVERED_LINES_KEY),
-        CoreMetrics.NEW_IT_LINE_COVERAGE_KEY);
-    }
-  }
-
-  private static class NewOverallLinesAndConditionsCoverageFormula extends NewLinesAndConditionsFormula {
-
-    private static final NewCoverageOutputMetricKeys OUTPUT_METRIC_KEYS = new NewCoverageOutputMetricKeys(
-      CoreMetrics.NEW_OVERALL_LINES_TO_COVER_KEY, CoreMetrics.NEW_OVERALL_UNCOVERED_LINES_KEY,
-      CoreMetrics.NEW_OVERALL_CONDITIONS_TO_COVER_KEY, CoreMetrics.NEW_OVERALL_UNCOVERED_CONDITIONS_KEY);
-    private static final Iterable<Formula<?>> VIEWS_FORMULAS = variationSumFormulas(OUTPUT_METRIC_KEYS);
-
-    private NewOverallLinesAndConditionsCoverageFormula(ScmInfoRepository scmInfoRepository) {
-      super(scmInfoRepository,
-        new NewCoverageInputMetricKeys(
-          CoreMetrics.OVERALL_COVERAGE_LINE_HITS_DATA_KEY, CoreMetrics.OVERALL_CONDITIONS_BY_LINE_KEY, CoreMetrics.OVERALL_COVERED_CONDITIONS_BY_LINE_KEY),
-        OUTPUT_METRIC_KEYS);
-    }
-
-    public static Iterable<Formula<?>> from(@Nullable ScmInfoRepository scmInfoRepository) {
-      if (scmInfoRepository == null) {
-        return VIEWS_FORMULAS;
-      }
-      return Collections.<Formula<?>>singleton(new NewOverallLinesAndConditionsCoverageFormula(scmInfoRepository));
-    }
-  }
-
-  private static class NewOverallCodeCoverageFormula extends LinesAndConditionsWithUncoveredVariationFormula {
-    public NewOverallCodeCoverageFormula() {
-      super(
-        new LinesAndConditionsWithUncoveredMetricKeys(
-          CoreMetrics.NEW_OVERALL_LINES_TO_COVER_KEY, CoreMetrics.NEW_OVERALL_CONDITIONS_TO_COVER_KEY,
-          CoreMetrics.NEW_OVERALL_UNCOVERED_LINES_KEY, CoreMetrics.NEW_OVERALL_UNCOVERED_CONDITIONS_KEY),
-        CoreMetrics.NEW_OVERALL_COVERAGE_KEY);
-    }
-  }
-
-  private static class NewOverallBranchCoverageFormula extends SingleWithUncoveredVariationFormula {
-    public NewOverallBranchCoverageFormula() {
-      super(
-        new SingleWithUncoveredMetricKeys(
-          CoreMetrics.NEW_OVERALL_CONDITIONS_TO_COVER_KEY, CoreMetrics.NEW_OVERALL_UNCOVERED_CONDITIONS_KEY),
-        CoreMetrics.NEW_OVERALL_BRANCH_COVERAGE_KEY);
-    }
-  }
-
-  private static class NewOverallLineCoverageFormula extends SingleWithUncoveredVariationFormula {
-    public NewOverallLineCoverageFormula() {
-      super(
-        new SingleWithUncoveredMetricKeys(
-          CoreMetrics.NEW_OVERALL_LINES_TO_COVER_KEY, CoreMetrics.NEW_OVERALL_UNCOVERED_LINES_KEY),
-        CoreMetrics.NEW_OVERALL_LINE_COVERAGE_KEY);
-    }
-  }
-
   /**
    * Creates a List of {@link org.sonar.server.computation.task.projectanalysis.formula.SumFormula.IntSumFormula} for each
    * metric key of the specified {@link NewCoverageOutputMetricKeys} instance.
    */
   private static Iterable<Formula<?>> variationSumFormulas(NewCoverageOutputMetricKeys outputMetricKeys) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this method into "NewLinesAndConditionsCoverageFormula". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS3398)